Spack provided nrnivmodl fails

```
$ nrnivmodl mod/
/p/project/cjzam11/kitayama1/projects/rhythms
Mod files: "mod/astrocyte.mod" "mod/Ca_conce.mod" "mod/cadecay_destexhe.mod" "mod/CALIca.mod" "mod/CALIkahp.mod" "mod/CALIkc.mod" "mod/CALIkdr.mod" "mod/CALIna.mod" "mod/cal.mod" "mod/halfgap.mod" "mod/hhr.mod" "mod/ICAN_destexhe.mod" "mod/IL_gutnick.mod" "mod/IM_cortex.mod" "mod/INTERIk.mod" "mod/INTERIna.mod" "mod/IT_huguenard.mod" "mod/kcRT03.mod" "mod/kdr.mod" "mod/nafPR.mod" "mod/passiv.mod" "mod/rcadecay.mod" "mod/rkq.mod" "mod/SinClamp.mod" "mod/test.mod" "mod/tN1fAstro.mod" "mod/tN1fN2.mod" "mod/tN2fAstro.mod" "mod/tN2fN1.mod"

Creating x86_64 directory for .o files.

COBJS=''
 -> Compiling mod_func.cpp
 -> NMODL ../mod//astrocyte.mod
 -> NMODL ../mod//Ca_conce.mod
 -> NMODL ../mod//cadecay_destexhe.mod
Spack compiler must be run from Spack! Input 'SPACK_ENV_PATH' is missing.
make: *** [/p/project/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.2.0/neuron-develop-4ktbfhjgbwghas6gfvknlxxlbredypsf/bin/nrnmech_makefile:120: mod_func.o] Error 1
make: *** Waiting for unfinished jobs....
Translating astrocyte.mod into /p/project/cjzam11/kitayama1/projects/rhythms/x86_64/astrocyte.c
Translating Ca_conce.mod into /p/project/cjzam11/kitayama1/projects/rhythms/x86_64/Ca_conce.c
Translating cadecay_destexhe.mod into /p/project/cjzam11/kitayama1/projects/rhythms/x86_64/cadecay_destexhe.c
Thread Safe
Thread Safe
Thread Safe
```

The NEURON is built with `spack install neuron@develop+mpi+python+coreneuron %gcc`

`nrnivmodl-core mod` fails:

```
$ nrnivmodl-core mod/
[INFO] Running: make -j4 -f /p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile ROOT=/p/project/cjzam11/kitayama1/opt/nrn-master MODS_PATH=x86_64/corenrn/mod2c BUILD_TYPE=SHARED
Default NMODL flags:
diff: x86_64/corenrn/mod2c/_mod_func.cpp: No such file or directory
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/CALIkc.cpp] Error 127
make: *** Waiting for unfinished jobs....
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/CALIkahp.cpp] Error 127
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/CALIca.cpp] Error 127
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/CALIkdr.cpp] Error 127
```

--- a/var/spack/repos/builtin/packages/neuron/package.py
+++ b/var/spack/repos/builtin/packages/neuron/package.py
@@ -104,7 +104,7 @@ class Neuron(CMakePackage):
                                      "./bin/nrnmech_makefile")
 
         # assign_operator is changed to fix wheel support
-        if self.spec.satisfies("+cmake") and self.spec.satisfies("@:7.99"):
+        if self.spec.satisfies("@:7.99"):
             assign_operator = "?="
         else:
             assign_operator = "="

--- a/var/spack/repos/builtin/packages/neuron/package.py
+++ b/var/spack/repos/builtin/packages/neuron/package.py
@@ -103,7 +103,12 @@ class Neuron(CMakePackage):
         nrnmech_makefile = join_path(self.prefix,
                                      "./bin/nrnmech_makefile")
 
-        assign_operator = "?="
+        # assign_operator is changed to fix wheel support
+        if self.spec.satisfies("+cmake") and self.spec.satisfies("@:7.99"):
+            assign_operator = "?="
+        else:
+            assign_operator = "="
+
         filter_file("CC {0} {1}".format(assign_operator, env["CC"]),
                     "CC = {0}".format(cc_compiler),
                     nrnmech_makefile,

